.github/workflows/build-docs.yml

```diff
@@ -0,0 +1,224 @@
+        run: |
+          git clone --depth 1 https://github.com/easyscience/EasyDiffractionLib .
+          git fetch --tags
+          echo "RELEASE_VERSION=$(git describe --tags --abbrev=0)" >> $GITHUB_ENV
+
+      # Activate dark mode to create documentation with Plotly charts in dark mode
+      # Need a better solution to automatically switch the chart colour theme based on the mkdocs material switcher
+      # Something similar to mkdocs_plotly_plugin https://haoda-li.github.io/mkdocs-plotly-plugin/,
+      # but for generating documentation from notepads
+      - name: Activate dark mode
+        run: |
+          brew install dark-mode
+          dark-mode status
+          dark-mode on
+          dark-mode status
+
+      - name: Check-out repository
+        uses: actions/checkout@v4
+
+      - name: Set up Python environment
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.12'
+
+      - name: Upgrade package installer for Python
+        run: python -m pip install --upgrade pip
+
+      # Install MkDocs -- static site generator
+      # https://www.mkdocs.org
+      - name: Install MkDocs and its dependencies
+        run: >
+          pip install mkdocs mkdocs-material 'mkdocs-autorefs<1.3.0'  
+          mkdocs-jupyter mkdocs-plugin-inline-svg
+          mkdocs-markdownextradata-plugin mkdocstrings-python
+
+      # Install EasyDiffraction Library to run Jupyter notebooks
+      # Install with the 'charts' and 'docs' extras
+      - name: Install EasyDiffraction Library and its dependencies
+        run: pip install 'easydiffraction[charts]'
+
+      # Download and add the extra files from the easyscience/assets-docs repository
+      - name: Get easyscience/assets-docs files
+        run: |
+          git clone https://github.com/easyscience/assets-docs.git
+          cp -R assets-docs/docs/assets/ docs/assets/
+          cp -R assets-docs/includes/ includes/
+          cp -R assets-docs/overrides/ overrides/
+
+      # Download and add the extra files from the easyscience/assets-branding repository
+      - name: Get easyscience/assets-branding files
+        run: |
+          git clone https://github.com/easyscience/assets-branding.git
+          mkdir -p docs/assets/images/
+          cp assets-branding/EasyDiffraction/logos/ed-logo_dark.svg docs/assets/images/
+          cp assets-branding/EasyDiffraction/logos/ed-logo_light.svg docs/assets/images/
+          cp assets-branding/EasyDiffraction/logos/edl-logo_dark.svg docs/assets/images/logo_dark.svg
+          cp assets-branding/EasyDiffraction/logos/edl-logo_light.svg docs/assets/images/logo_light.svg
+          cp assets-branding/EasyDiffraction/icons/ed-icon_256x256.png docs/assets/images/favicon.png
+          mkdir -p overrides/.icons/
+          cp assets-branding/EasyDiffraction/icons/ed-icon_bw.svg overrides/.icons/easydiffraction.svg
+          cp assets-branding/EasyScience/icons/es-icon_bw.svg overrides/.icons/easyscience.svg
```

Comment:
These preparatory steps are present in `tools/prepare_docs.sh` - why not just run this script here?
(and for all other steps, which have corresponding scripts)

Reply:
Check out my reply above.